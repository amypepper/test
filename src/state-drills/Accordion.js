import React from "react";

class Accordion extends React.Component {
  static defaultProps = { sections: [] };

  constructor(props) {
    super(props);
    this.state = {
      activeSectionIndex: null,
    };
  }

  handleButtonClick(index) {
    this.setState({ activeSectionIndex: index });
  }
  renderLis(section, index, activeSectionIndex) {
    return (
      <li key={index}>
        <button onClick={() => this.handleButtonClick(index)}>
          {section.title}
        </button>
        {activeSectionIndex === index && <p>{section.content}</p>}
      </li>
    );
  }
  render() {
    console.log(this.state.activeSectionIndex);
    return (
      <ul>
        {this.props.sections.map((section, index) =>
          this.renderLis(section, index, this.state.activeSectionIndex)
        )}
      </ul>
    );
  }
}
export default Accordion;
