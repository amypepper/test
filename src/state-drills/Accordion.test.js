import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Accordion from "./Accordion";
import sections from "./sections";

describe("Accordion component", () => {
  it("renders without errors", () => {
    const div = document.createElement("div");
    ReactDOM.render(<Accordion />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
  it("renders empty given no prop", () => {
    const wrapper = shallow(<Accordion />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it("renders no sections as active by default", () => {
    const wrapper = shallow(<Accordion sections={sections} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it("opens a clicked section", () => {
    const wrapper = shallow(<Accordion sections={sections} />);
    wrapper.find("button").at(1).simulate("click");
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it("only opens the most recently clicked section", () => {
    const wrapper = shallow(<Accordion sections={sections} />);
    wrapper.find("button").at(2).simulate("click");
    wrapper.find("button").at(0).simulate("click");
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
