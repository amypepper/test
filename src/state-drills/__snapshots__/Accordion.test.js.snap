// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion component only opens the most recently clicked section 1`] = `
<ul>
  <li
    key="0"
  >
    <button
      onClick={[Function]}
    >
      Section 1
    </button>
    <p>
      Lorem ipsum dolor sit amet consectetur adipisicing elit.
    </p>
  </li>
  <li
    key="1"
  >
    <button
      onClick={[Function]}
    >
      Section 2
    </button>
  </li>
  <li
    key="2"
  >
    <button
      onClick={[Function]}
    >
      Section 3
    </button>
  </li>
</ul>
`;

exports[`Accordion component opens a clicked section 1`] = `
<ul>
  <li
    key="0"
  >
    <button
      onClick={[Function]}
    >
      Section 1
    </button>
  </li>
  <li
    key="1"
  >
    <button
      onClick={[Function]}
    >
      Section 2
    </button>
    <p>
      Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!
    </p>
  </li>
  <li
    key="2"
  >
    <button
      onClick={[Function]}
    >
      Section 3
    </button>
  </li>
</ul>
`;

exports[`Accordion component renders empty given no prop 1`] = `<ul />`;

exports[`Accordion component renders no sections as active by default 1`] = `
<ul>
  <li
    key="0"
  >
    <button
      onClick={[Function]}
    >
      Section 1
    </button>
  </li>
  <li
    key="1"
  >
    <button
      onClick={[Function]}
    >
      Section 2
    </button>
  </li>
  <li
    key="2"
  >
    <button
      onClick={[Function]}
    >
      Section 3
    </button>
  </li>
</ul>
`;
